azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/AvailabilitySet.java

```diff
@@ -64,27 +64,42 @@
 
 
     /**************************************************************
-     * Fluent interfaces to provision an AvailabilitySet
+     * Fluent interfaces to create an AvailabilitySet
      **************************************************************/
 
     /**
+     * Container interface for all the definitions
+     */
+    public interface Definitions extends
+        DefinitionBlank,
+        DefinitionWithGroup,
+        DefinitionAfterGroup,
+        DefinitionCreatable {
+    }
+    
+    /**
      * The first stage of an availability set definition
      */
-    interface DefinitionBlank extends GroupableResource.DefinitionWithRegion<DefinitionWithGroup> {
+    public interface DefinitionBlank extends GroupableResource.DefinitionWithRegion<DefinitionWithGroup> {
```

Comment:
nope. 
public is inferred on the *members* of an interface, not on the interface itself
https://docs.oracle.com/javase/tutorial/java/IandI/interfaceDef.html